igor-web/src/main/groovy/com/netflix/spinnaker/igor/scm/KeelManifestsController.groovy

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.igor.scm
+
+import com.netflix.spinnaker.igor.scm.bitbucket.client.BitBucketMaster
+import com.netflix.spinnaker.igor.scm.github.client.GitHubMaster
+import com.netflix.spinnaker.igor.scm.gitlab.client.GitLabMaster
+import com.netflix.spinnaker.igor.scm.stash.client.StashMaster
+import com.netflix.spinnaker.igor.scm.stash.client.model.DirectoryListingResponse
+import com.netflix.spinnaker.igor.scm.stash.client.model.TextLinesResponse
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import retrofit.RetrofitError
+
+/**
+ * Exposes APIs to retrieve keel declarative manifests from source control repos.
+ */
+@RestController
+@Slf4j
+@RequestMapping("/keel")
+class KeelManifestsController {
+  @Autowired(required = false)
+  StashMaster stashMaster
+
+  @Autowired(required = false)
+  GitHubMaster gitHubMaster
+
+  @Autowired(required = false)
+  GitLabMaster gitLabMaster
+
+  @Autowired(required = false)
+  BitBucketMaster bitBucketMaster
+
+  static final String KEEL_MANIFESTS_BASE_PATH = ".netflix/spinnaker"
```

Comment:
Instead of allowing users to specify the manifest directory, I feel like we should enforce a convention, whatever it is.

Reply:
I think all the things on the igor side should be generic.

Reply:
There could be a default in the igor config if we decide it should be set on the igor side (maybe that's the best idea to limit exposure)?  but I think it should default to `.spinnaker/`

Reply:
I like `.spinnaker`, but was following the example of `keeldemo`. Also, I thought when we saw the `newt` demo earlier today, the file with declarative configs was `.netflix/spinnaker.yml`? I don't have a strong opinion one way or another -- we just need to settle on _one_ convention. 😃

Reply:
It should be configurable in the igor config regardless. `.netflix/` will be what it is internal to netflix, but that shouldn't be the default for oss. I think the default for oss should be `.spinnaker/`